Gems/ROS2/Code/Source/SimulationUtils/FollowingCameraComponent.cpp

```diff
@@ -170,93 +194,323 @@ namespace ROS2
+                AzFramework::InputSystemCursorRequestBus::Event(
+                    AzFramework::InputDeviceMouse::Id,
+                    &AzFramework::InputSystemCursorRequests::SetSystemCursorPositionNormalized,
+                    m_initialMousePosition);
+
+                // Update m_lastMousePosition to the restored position to prevent the jump on the next rotation start
+                m_lastMousePosition = m_initialMousePosition;
+            }
         }
-        if (channelId == AzFramework::InputDeviceKeyboard::Key::AlphanumericS)
+
+        if (m_isRightMouseButtonPressed &&
+            (channelId == AzFramework::InputDeviceMouse::Movement::X || channelId == AzFramework::InputDeviceMouse::Movement::Y))
         {
-            m_opticalAxisTranslation -= m_configuration.m_zoomSpeed;
-            return;
+            AZ::Vector2 currentMousePosition = GetCurrentMousePosition();
+            AZ::Vector2 mouseDelta = currentMousePosition - m_lastMousePosition;
+
+            RotateCameraOnMouse(mouseDelta);
+
+            const auto center = AZ::Vector2(0.5f, 0.5f);
+
+            // Recenter the cursor to avoid edge constraints
+            AzFramework::InputSystemCursorRequestBus::Event(
+                AzFramework::InputDeviceMouse::Id, &AzFramework::InputSystemCursorRequests::SetSystemCursorPositionNormalized, center);
+
+            // Then update m_lastMousePosition accordingly
+            m_lastMousePosition = center;
         }
-        if (channelId == AzFramework::InputDeviceKeyboard::Key::AlphanumericA)
+    }
+
+    AZ::Vector2 FollowingCameraComponent::GetCurrentMousePosition()
+    {
+        AZ::Vector2 mousePosition(0.0f, 0.0f);
+
+        // Query the input system for the current mouse position
+        AzFramework::InputSystemCursorRequestBus::EventResult(
+            mousePosition, AzFramework::InputDeviceMouse::Id, &AzFramework::InputSystemCursorRequests::GetSystemCursorPositionNormalized);
+
+        return mousePosition;
+    }
+
+    void FollowingCameraComponent::RotateCameraOnMouse(const AZ::Vector2& mouseDelta)
+    {
+        const float rotationSensitivity = m_configuration.m_rotationSensitivity;
+
+        // Convert mouse delta to rotation angles
+        float yawRotation = -mouseDelta.GetX() * rotationSensitivity; // Inverted X for reversed yaw rotation
+        float pitchRotation = -mouseDelta.GetY() * rotationSensitivity; // Inverted Y for pitch, adjust as needed
+
+        // Fetch the current camera orientation from m_cameraOffset
+        AZ::Quaternion currentRotation = AZ::Quaternion::CreateFromMatrix4x4(m_cameraOffset);
+
+        // Global Z-axis for yaw rotation
+        AZ::Vector3 globalZAxis = AZ::Vector3(0, 0, 1);
+        AZ::Quaternion yawQuat = AZ::Quaternion::CreateFromAxisAngle(globalZAxis, yawRotation);
+
+        // Apply yaw rotation to current camera orientation
+        AZ::Quaternion tempRotation = yawQuat * currentRotation;
+        tempRotation.Normalize();
```

Comment:
I think you need no normalization, since yawQuat and currentRotation. If those are not normalized is not ok.
I would suggest assert on IsNormalized.